app/assets/javascripts/services/ChatStates.ts

```diff
@@ -250,6 +250,7 @@ export class EngagedState {
     }
```

Comment:
This is a temporary fix which makes escalated true if the agent is typing. This will direct people using webchats to the _sendPostChatSurveyWebchat() method. Previously, this was not happening. This will cover all situations such as if they are in a webchat and there is no typing from an agent before it closes.

When auto engagement is set up, we should try and link it to that.